packages/sitecore-jss-nextjs/src/services/component-props-service.test.ts

```diff
@@ -0,0 +1,309 @@
+import { ComponentRendering, PlaceholdersData } from '@sitecore-jss/sitecore-jss';
+import { expect, use, spy } from 'chai';
+import spies from 'chai-spies';
+import {IncomingMessage, ServerResponse} from 'http';
+import {ParsedUrlQuery} from 'querystring';
+import { ComponentPropsService, ComponentPropsRequest } from './component-props-service';
+
+use(spies);
+
+describe('ComponentPropsService', () => {
+  const service = new ComponentPropsService();
+
+  const rendering = (name: string): ComponentRendering => ({
+    uid: name,
+    componentName: name,
+  });
+
+  type CustomContext = { locale: string };
+
+  const placeholders = {
```

Comment:
How are we handling the same component/rendering added multiple times (to the same placeholder, or spread among different placeholders)? I think these are valid scenarios - the renderings may differ by `datasource` or `params` - so you'd have multiple renderings with the same componentName, but different uid. Might be good to have a test for this scenario.

Reply:
Now I use uid in order to store data, instead of componentName.

Reply:
this `placeholders` - contains MyCustomComponent - several times, and in different placeholders